docs/wiki/deployment/performance-safety.md

```diff
@@ -59,6 +59,49 @@ Profiling query: SELECT DISTINCT process.name, listening.port, listening.protoco
+
+The script returns 5 stats:
+
+**Utilization (U)**: Utilization is calculated by taking the average of non-0 results of the cpu_percent(interval=1) function in `psutils.Process()`. This value can be greater than 100% for processes with threads running across different CPUs. The script sets an interval of 1 meaning that the function compares process time to system CPU times elapsed before and after the 1 second interval. This is a blocking call. 
+
+**CPU time (C)**: CPU time uses the `psutils.Process()`'s `cpu_times()` function. It returns a named tuple containing user, system, children_user, system_user, and iowait
+
+- user: time spent in user mode.
+- system: time spent in kernel mode.
+- children_user: user time of all child processes (always 0 on Windows and macOS).
+- system_user: user time of all child processes (always 0 on Windows and macOS).
+- iowait: (Linux) time spent waiting for blocking I/O to complete. This value is excluded from user and system times count (because the CPU is not doing any work).
+
+The profile script adds user and system together for the CPU 
+Time output.
+
+**Duration (D)**:
+Duration is calculated by taking the subtracting `start_time` - 2 from the current time. The start time is set before the script starts the `osqueryi` process to run the query. The `start_time` - 2 comes from the `--profile_delay` flag used by the profile.py script. This flag causes osquery to wait before and after running the code under test. The 2 is to make up for this wait time. 
+
+**fds (F)**: Uses the `num_fds()` function and returns the file descriptors used by the `osqueryi` process during query execution
+
+**Memory (M)**: Uses the `memory_info_ex()` function which is deprecated. psutils documentation suggests using `memory_info()` instead. The function returns a named tuple and the script uses the `rss` value in the tuple. RSS stands for resident set size and is the non-swapped physical memory used by the process. This should match the RES column in `top`.
+
+**Profile.py Simple Thresholds**
+
+The numbers next to the stats in the script output are determined by the `RANGES` dictionary in `profile.py`
+
+```
+KB = 1024 * 1024
+RANGES = {
+    "colors": (utils.blue, utils.green, utils.yellow, utils.red),
+    "utilization": (8, 20, 50),
+    "cpu_time": (0.4, 1, 10),
+    "memory": (8 * KB, 12 * KB, 24 * KB),
+    "fds": (10, 20, 50),
+    "duration": (0.8, 1, 3),
+}
+```
+
+The script will take the value of the stat and compare it with the tuple at the corresponding stat's key in `RANGES`. If the value is less than the value in the tuple then the index for the value in the tuple is what appears in the script output. If the value for the stat is greater than all values of the tuple, then the length of the tuple is what appears in the script output. For example, if `cpu_time` for a query is 0.2, then you'll see `C: 0` in the script output. If `cpu_time` is 11, then you'll see `C:3` in the script output. 
+
```

Comment:
The words below ("The results (utilization=2) suggest running") make me wonder if this section should be down a couple paragraphs?

Reply:
I'll drop it down to be above "Queries that fail to execute...". I'll also update the language to be more consistent with the existing docs, using "category" to describe the thresholds.